### PR TITLE
Ekf config parser update

### PR DESCRIPTION
**Pull request by PNieck**
## Description

This PR changes how the configuration files are read. Now parser library is used.

## Motivation and Context

1. Removing old temporary solution.
2. Making config reads more consistent

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here). `armv7a9-zynq7000-qemu`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
https://github.com/phoenix-pilot/phoenix-pilot-project/pull/42
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: libs/parser/parser.h [hunk 1]
```diff
@@ -49,4 +52,8 @@ extern int parser_execute(parser_t *p, const char *path, unsigned int mode);
 extern void parser_clear(parser_t *p);
 
 
+/* Function parses field from `h` with name `fieldName` to variable called `target`. `fieldType` defines what is the expected type of this field. */
+extern int parser_fieldParseHelper(const hmap_t *h, const char *fieldName, void *target, parser_fieldType fieldType);
```
Comment (niewim19) [inline libs/parser/parser.h:56]: This function is ought to be used inside converters - please mention it in the comment

File: ekf/kalman_implem.c [hunk 2]
```diff
@@ -37,73 +41,106 @@ struct {
...
-					break;
-				}
-			}
-		}
-		fclose(fd);
+
+	err |= parser_headerAdd(p, "P_MATRIX", kmn_PMatrixConverter);
+	err |= parser_headerAdd(p, "R_MATRIX", kmn_RMatrixConverter);
+	err |= parser_headerAdd(p, "Q_MATRIX", kmn_QMatrixConverter);
+	err |= parser_headerAdd(p, "LOGGING", kmn_loggingConverter);
```
Comment (niewim19) [inline ekf/kalman_implem.c:129]: Just an idea for the future:

parser_headerAdd can accept a void pointer that will be passed to converters as target space to write out data. This way parser would be more thread safe as now it heavily rely on common structures to write out data.

File: ekf/kalman_implem.c [hunk 3]
```diff
@@ -37,73 +41,106 @@ struct {
...
-	.R_bwstdev = 0.0001, /* standard deviation of gyroscope bias estimation in radians */
+static int kmn_PMatrixConverter(const hmap_t *h)
+{
+	int err = 0;
 
-	.R_hstdev = 0.025, /* standard deviation of change in height in meters */
+	err |= parser_fieldParseHelper(h, "P_qerr", &converterResult->P_qerr, parser_float);
+	err |= parser_fieldParseHelper(h, "P_verr", &converterResult->P_verr, parser_float);
+	err |= parser_fieldParseHelper(h, "P_baerr", &converterResult->P_baerr, parser_float);
+	err |= parser_fieldParseHelper(h, "P_rerr", &converterResult->P_rerr, parser_float);
```
Comment (niewim19) [inline ekf/kalman_implem.c:54]: As we discussed the prefix `P_`/`Q_` may be dropped in the config file. The header specifies the context.

File: ekf/kalman_implem.h [hunk 4]
```diff
@@ -126,7 +125,8 @@ typedef struct {
 int verbose;
 
 
-extern void kmn_configRead(kalman_init_t *initVals);
+/* Function reads kalman config file and fills  */
+extern int kmn_configRead(const char *configFile, kalman_init_t *initVals);
```
Comment (niewim19) [inline ekf/kalman_implem.h:129]:
```suggestion
/* Function reads ekf configuration file under `path` and fills structure pointed by `initVals`  */
extern int kmn_configRead(const char *path, kalman_init_t *initVals);
```

**Review by niewim19 — CHANGES_REQUESTED** [review]
Initially it was assumed that a separate file for field interpreting functions would be needed. As it turns out this does not need so much code so I think it may be bundled into one function and put into the `parser.c` file.

File: libs/parser/converter_helpers.c [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * %LICENSE%
+ */
+
+#include "parser.h"
+
+#include <stdio.h>
+
+
+union parser_target {
```
[the remark this answers is not in the document]
Comment (PNieck) [reply libs/parser/converter_helpers.c:19]: When I use `static` I get `useless storage class specifier in empty declaration` compilation error

File: ekf/kalman_implem.c (hunk 2 above)
Comment (PNieck) [reply ekf/kalman_implem.c:129]: Good idea, I have saved it in issue connected with parser refactoring

https://github.com/phoenix-pilot/phoenix-pilot-core/issues/197

File: libs/parser/parser.c [hunk 6]
```diff
@@ -451,3 +448,35 @@ __attribute__((destructor)) static void parser_regexFree(void)
 		regfree(&parser_common.headerNameRegex);
 	}
 }
+
+
+int parser_fieldGet(const hmap_t *h, const char *fieldName, void *target, parser_fieldType fieldType)
+{
+	char *valueStr, *endptr;
+
```
Comment (niewim19) [inline libs/parser/parser.c:456]: Please add NULL check for `target`

**Review by niewim19 — COMMENTED** [review]
Only one small change and lets go

File: libs/parser/parser.c (hunk 6 above)
Comment (kemonats) [reply libs/parser/parser.c:456]: And maybe aligning the `target` address to 4

Comment (PNieck) [reply libs/parser/parser.c:456]: @kemonats You mean checking if `target % 4 == 0`, am I right?

Comment (PNieck) [reply libs/parser/parser.c:456]: Done. I have fixed as the @agkaminski suggested.

**Pull request merged**